### PR TITLE
Start on some new How Tos on JS testing

### PR DESCRIPTION
**Pull request by jim**
## Description

I had to look up some commands while working on another PR, so I added them to our How Tos.

## Reviewer Notes

* I would like to err on the side of merging soon and using future PRs (authored by myself or, hopefully, others) to iteratively improve to the approaches described. If, however, something in here is a Bad Idea™, let's definitely take it out before merging.

### CODE REVIEW

File: docs/how-to/run-e2e-tests.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# How To Run End to End (Cypress) Tests
+
+## Run All End to End Tests
+
+```console
+$ make e2e_test
+```
+
```
Comment (kahlouie) [inline docs/how-to/run-e2e-tests.md]: can we add the commands to regenerate the db and restart the tests? That's been super handy for me.

File: docs/how-to/unit-test-react-components.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    expect(wrapper.containsMatchingElement(Alert)).toEqual(false);
+  });
+});
+```
+
+Alternatively, you can use `html()` to render a node to HTML and make assertions against the markup.
+
+## Full Rendering
+
+While providing a more realistic perspective on the application's behavior, using `mount` can be problematic. One cause of this is that certain components (such as `Link` from `react-router`) can't be rendered unless they appear beneath another component or provider. In such a case, you'll need to setup any providers or parent components and render the component under test inside them.
```
Comment (kahlouie) [inline docs/how-to/unit-test-react-components.md]: This last sentence is a bit hard to parse. Can you please elaborate or refine?

**Comment by Ronolibert** [comment]
I'm not sure where I would add it, but I would add a note that using `make e2e_test` uses the data generated from the `e2ebasic.go` file. So when we want to add test cases to the e2e tests we add another scenario to that file since some cypress test cases mutate the app state

**Comment by jim** [comment]
@Ronolibert I'll add that to my list of things to cover in a future how-to. I'm thinking one about how to add an e2e test could mention how to view/modify the data that is in that scenario.
